File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.cpp
class Solution {
public:
    vector<int> solve(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> mp;
        vector<int> res;
        for (int i: nums2) {
            mp[i]++;
        }
        for (int i: nums1) {
            if (mp.count(i) == 0) {
                res.push_back(i);
                mp[i]++;
            }
        }
        return res;
    }
    
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        vector<vector<int>> res(2);
        res[0] = solve(nums1, nums2);
        res[1] = solve(nums2, nums1);
        return res;
    }
};

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        List<List<Integer>> result = new ArrayList<>();
        result.add(getDifference(nums1, nums2));
        result.add(getDifference(nums2, nums1));
        return result;
    }
    
    public List<Integer> getDifference(int[] nums1, int[] nums2) {
        Set<Integer> set = new HashSet<>();
        for (int i = 0; i < nums2.length; i++) {
            set.add(nums2[i]);
        }
        List<Integer> list = new ArrayList<>();
        for (int i = 0; i < nums1.length; i++) {
            if (!set.contains(nums1[i])) {
                list.add(nums1[i]);
            }
            set.add(nums1[i]);
        }
        return list;
    }
}